crates/edr_napi/src/mock/time.rs

```diff
@@ -0,0 +1,101 @@
+use std::sync::Arc;
+
+use edr_generic::GenericChainSpec;
+use edr_napi_core::logger::Logger;
+use edr_solidity::contract_decoder::ContractDecoder;
+use napi::{bindgen_prelude::BigInt, tokio::runtime, Env};
+use napi_derive::napi;
+
+use crate::{
+    cast::TryCast as _,
+    config::{ProviderConfig, TracingConfigWithBuffers},
+    logger::LoggerConfig,
+    provider::Provider,
+    subscription::SubscriptionConfig,
+};
+
+#[napi]
+pub struct MockTime {
+    inner: Arc<edr_provider::time::MockTime>,
+}
+
+#[napi]
+impl MockTime {
+    #[doc = "Creates a new instance of `MockTime` with the current time."]
+    #[napi(factory)]
+    pub fn now() -> Self {
+        Self {
+            inner: Arc::new(edr_provider::time::MockTime::now()),
+        }
+    }
+
+    #[doc = "Adds the specified number of seconds to the current time."]
+    #[napi]
+    pub fn add_seconds(&self, seconds: BigInt) -> napi::Result<()> {
+        let seconds = seconds.try_cast()?;
+
+        self.inner.add_seconds(seconds);
+        Ok(())
+    }
+}
+
+#[doc = "Creates a provider with a mock timer."]
+#[doc = "For testing purposes."]
+#[napi]
+pub fn create_provider_with_mock_timer(
```

Comment:
Is this function supposed to be shipped with the library?

Reply:
The whole `mod mock` (one level up) is only included if the `test-mock` feature flag is enabled. As a result, this module will never be compiled.